lib/std/Build.zig

```diff
@@ -1596,6 +1598,45 @@ pub const Module = struct {
     /// file of directory of files which constitute the module.
     source_file: FileSource,
     dependencies: std.StringArrayHashMap(*Module),
+    include_dirs: std.ArrayListUnmanaged(CompileStep.IncludeDir) = .{},
+    lib_paths: std.ArrayListUnmanaged([]const u8) = .{},
+    system_libs: std.ArrayListUnmanaged([]const u8) = .{},
+    libs: std.ArrayListUnmanaged(*Build.CompileStep) = .{},
+    config_headers: std.ArrayListUnmanaged(*Build.ConfigHeaderStep) = .{},
+    link_libc: bool = false,
+    link_libcpp: bool = false,
+
+    pub fn addIncludePath(m: *Module, path: []const u8) void {
+        m.include_dirs.append(
+            m.builder.allocator,
+            CompileStep.IncludeDir{ .raw_path = m.builder.dupe(path) },
+        ) catch @panic("OOM");
+    }
+
+    pub fn addConfigHeader(m: *Module, config_header: *Build.ConfigHeaderStep) void {
+        m.config_headers.append(config_header) catch @panic("OOM");
+    }
+
+    pub fn addLibraryPath(m: *Module, library_path: []const u8) void {
+        m.lib_paths.append(m.builder.dupe(library_path)) catch @panic("OOM");
+    }
+
+    pub fn linkLibC(m: *Module) void {
+        m.link_libc = true;
+    }
+
+    pub fn linkLibCpp(m: *Module) void {
+        m.link_libcpp = true;
+    }
+
+    pub fn linkLibrary(m: *Module, lib: *Build.CompileStep) void {
+        m.libs.append(m.builder.allocator, lib) catch @panic("OOM");
+    }
+
+    pub fn linkSystemLibrary(m: *Module, lib: []const u8) void {
+        // TODO: Ignoring `needed` and `weak` variants for now.
+        m.system_libs.append(m.builder.allocator, m.builder.dupe(lib)) catch @panic("OOM");
```

Comment:
This technically makes `link_libc` and `link_libcpp` redundant since they're equivalent to `linkSystemLibrary("c")` and `linkSystemLibrary("c++")` respectively (see `CompileStep.isLibCLibrary`). To make this a bit clearer, maybe have `Module.linkLibC[pp]` just call `linkSystemLibrary`, to stop there being two ways to represent this linking behaviour?

Reply:
Fixed